ntbs-service-unit-tests/DataMigration/CaseManagerImportServiceTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Moq;
+using ntbs_service.DataAccess;
+using ntbs_service.DataMigration;
+using ntbs_service.DataMigration.RawModels;
+using ntbs_service.Models.Entities;
+using ntbs_service.Models.ReferenceEntities;
+using Xunit;
+
+namespace ntbs_service_unit_tests.DataMigration
+{
+    public class CaseManagerImportServiceTests : IClassFixture<CaseManagerImportServiceTests.DatabaseFixture>
+    {
+        private readonly CaseManagerImportService _caseManagerImportService;
+        private readonly Mock<IMigrationRepository> _migrationRepositoryMock = new Mock<IMigrationRepository>();
+
+        private Dictionary<string, IEnumerable<MigrationDbNotification>> _idToNotificationDict;
+        private Dictionary<string, MigrationLegacyUser> _usernameToLegacyUserDict;
+        private Dictionary<string, IEnumerable<MigrationLegacyUserHospital>> _usernameToLegacyUserHospitalDict;
+        private readonly NtbsContext _context;
+
+        public CaseManagerImportServiceTests(DatabaseFixture fixture)
+        {
+            _context = fixture.Context;
+            SetupMockMigrationRepo();
+            IUserRepository userRepository = new UserRepository(_context);
+            IReferenceDataRepository referenceDataRepository = new ReferenceDataRepository(_context);
+            var importLogger = new ImportLogger();
+            _caseManagerImportService = new CaseManagerImportService(userRepository, referenceDataRepository,
+                _migrationRepositoryMock.Object, importLogger);
+        }
+
+        [Fact]
+        public async Task WhenCaseManagerForLegacyNotificationDoesNotExistInNtbs_AddsCaseManager()
+        {
+            // Arrange
+            var notification = new Notification
+            {
+                IsLegacy = true,
+                LTBRID = "11111",
+                HospitalDetails = new HospitalDetails {TBServiceCode = "TBS00TEST"}
+            };
+            _idToNotificationDict = new Dictionary<string, IEnumerable<MigrationDbNotification>>
+            {
+                {
+                    "11111",
+                    new List<MigrationDbNotification> {new MigrationDbNotification {CaseManager = "TestUser@nhs.net"}}
+                }
+            };
+            _usernameToLegacyUserDict = new Dictionary<string, MigrationLegacyUser>
+            {
+                {
+                    "TestUser@nhs.net",
+                    new MigrationLegacyUser {Username = "TestUser@nhs.net", GivenName = "John", FamilyName = "Johnston"}
+                }
+            };
+            _usernameToLegacyUserHospitalDict = new Dictionary<string, IEnumerable<MigrationLegacyUserHospital>> 
+                {{ "TestUser@nhs.net", new List<MigrationLegacyUserHospital>() }};
```

Comment:
I think you could both (a) make your tests easier to understand, and (b) commonise the code a bit more, if you made them a bit more declarative. For example, you could have methods like `SetupLegacyNotification`, `SetupLegacyUser` etc. and you don't necessarily have to pass all this information in (e.g. in these tests the user always has the same name, so this could be hard-coded in the method, rather than needing to pass in as variables).

There's no specific action here (and if you choose to leave this as-is for now then that's ok) but going forward it would be good to think about writing tests in such a way that it's obvious to readers what the test is testing.